src/colorizer/Collection.ts

```diff
@@ -235,42 +243,70 @@ export default class Collection {
-   * @param fetchMethod Optional. The fetch command used to retrieve the URL.
+   * @param options Optional configuration, containing any of the following properties:
+   * - `fetchMethod` optional override for the fetch method, used to retrieve the URL.
+   * - `reportWarning` optional callback for reporting warning messages during loading.
    * @throws Error if the JSON could not be retrieved or is an unrecognized format.
    * @returns A new Collection object containing the retrieved data.
    */
-  public static async loadCollection(collectionParam: string, fetchMethod = fetchWithTimeout): Promise<Collection> {
+  public static async loadCollection(
+    collectionParam: string,
+    options: Partial<{ fetchMethod: typeof fetchWithTimeout; reportWarning: ReportWarningCallback }> = {}
+  ): Promise<Collection> {
     const absoluteCollectionUrl = Collection.formatAbsoluteCollectionPath(collectionParam);
 
     let response;
     try {
+      const fetchMethod = options.fetchMethod ?? fetchWithTimeout;
       response = await fetchMethod(absoluteCollectionUrl, DEFAULT_FETCH_TIMEOUT_MS);
     } catch (e) {
-      throw new Error(`Could not retrieve collections JSON data from url '${absoluteCollectionUrl}': '${e}'`);
+      throw new Error(
+        LoadErrorMessage.UNREACHABLE_COLLECTION +
+          " This may be due to a network issue, the server being unreachable, or a misconfigured URL." +
+          " Please check your network access."
+      );
     }
     if (!response.ok) {
-      throw new Error(`Could not retrieve collections JSON data from url '${absoluteCollectionUrl}': Fetch failed.`);
+      console.error(`Failed to fetch collections JSON from url '${absoluteCollectionUrl}':`, response);
+      throw new Error(
+        `Received a ${response.status} (${response.statusText}) code from the server while retrieving collections JSON from url '${absoluteCollectionUrl}'.`
+      );
     }
 
     let collection: CollectionFile;
     try {
       const json = await response.json();
       collection = updateCollectionVersion(json);
     } catch (e) {
-      throw new Error(`Could not parse collections JSON data from url '${absoluteCollectionUrl}': '${e}'`);
+      throw new Error(
+        `Parsing failed for the collections JSON file with the following error. Please check that the JSON syntax is correct: ${e}`
+      );
     }
 
     // Convert JSON array into map
-    if (collection.datasets.length === 0) {
+    if (!collection.datasets || collection.datasets.length === 0) {
       throw new Error(
-        `Could not retrieve collections JSON data from url '${absoluteCollectionUrl}': No datasets found.`
+        "Collection JSON was loaded but no datasets were found. At least one dataset must be defined in the collection."
       );
     }
     const collectionData: Map<string, CollectionEntry> = new Map();
+    const duplicateDatasetNames = new Set<string>();
```

Comment:
For just organization I suggest factoring this out into a `checkForDuplicateDatasetNames` since this is already really long. Personally, I would probably have a file that has all the error messages and names so you could do stuff like `throw new Error(NO_DATASETS_FOUND)`